clx/analytics/detector_utils.py

```diff
@@ -0,0 +1,91 @@
+import rmm
+import cudf
+import logging
+import numpy as np
+
+log = logging.getLogger(__name__)
+
+
+class DetectorDataset(object):
```

Comment:
Hi Bhargav, I'm concerned this class wrapper might be too thin of a wrapper holding just a list and count. Is there a way we can leverage this class a bit more? The utility functions seem useful, can we move some of that functionality to this class? Let me know if you want to discuss.

Reply:
Hi Bianca, Yea, I agree. We can have offline discussion.
I am thinking of creating private constructor for `DetectorDataset` and move `prepare_detector_dataset` function to DetectorDataset and make it as static. So, that dataset object can only be initiate with `prepare_detector_dataset `. 
do you think this is a good idea?

Reply:
@brhodes10 @efajardo-nv I have updated dga files as per the suggestion. thank you!